wandb/sdk/lib/sock_client.py

```diff
@@ -149,20 +153,48 @@ def send_record_publish(self, record: "pb.Record") -> None:
         server_req.record_publish.CopyFrom(record)
         self.send_server_request(server_req)
 
+    def _buffer_get(self, start: int, end: int, advance: bool = False) -> bytes:
+        if advance:
+            data = b"".join(self._buffer_list)
```

Comment:
once done with it, explicitly del it to help garbage collector?

Reply:
i just didnt want to deal with the maintenance of the buffer_list once partial data was extracted from it.. but that is what is needed.